File: frontend/vue-vben-admin/apps/web-antd/src/api/system/department.ts
import { requestClient } from '#/api/request';

export interface DepartmentItem {
  children: DepartmentItem[];
  code: string;
  id: string;
  isEnabled: boolean;
  leader?: null | string;
  name: string;
  order: number;
  parentId?: null | string;
  phone?: null | string;
}

export interface SaveDepartmentParams {
  code: string;
  isEnabled: boolean;
  leader?: null | string;
  name: string;
  order: number;
  parentId?: null | string;
  phone?: null | string;
}

export async function getDepartmentListApi() {
  return requestClient.get<DepartmentItem[]>('/system/department/list');
}

export async function createDepartmentApi(data: SaveDepartmentParams) {
  return requestClient.post<DepartmentItem>('/system/department', data);
}

export async function updateDepartmentApi(
  id: string,
  data: SaveDepartmentParams,
) {
  return requestClient.put<DepartmentItem>(`/system/department/${id}`, data);
}

export async function deleteDepartmentApi(id: string) {
  return requestClient.delete<boolean>(`/system/department/${id}`);
}

File: src/MiniAdmin.Domain/Entities/Department.cs
namespace MiniAdmin.Domain.Entities;

public sealed class Department
{
    public Guid Id { get; set; }

    public Guid? TenantId { get; set; }

    public Tenant? Tenant { get; set; }

    public Guid? ParentId { get; set; }

    public Department? Parent { get; set; }

    public string Code { get; set; } = string.Empty;

    public string Name { get; set; } = string.Empty;

    public string? Leader { get; set; }

    public string? Phone { get; set; }

    public int Order { get; set; }

    public bool IsEnabled { get; set; } = true;

    public List<Department> Children { get; set; } = [];
}
